File: vol1/ch11/LeetCode325.cpp
class Solution {
public:
    int maxSubArrayLen(vector<int>& nums,int k) {
        unordered_map<long,long> hmap;
        hmap[0]=-1;
        long presum=0,ans=0; 
        for (int j=0;j<nums.size();j++){
            presum+=nums[j];
            long rest=presum-k;
            if (hmap.count(rest)>0) 
                ans=max(ans,j-hmap[rest]); 
            if(hmap.count(presum)==0)
                hmap[presum]=j;
        } 
        return ans; 
    }
};

File: vol1/ch11/LeetCode325.py
class Solution:
    def maxSubArrayLen(self, nums: List[int], k: int) -> int:
        hmap={}
        hmap[0]=-1
        presum,ans=0,0 
        for j in range(0,len(nums)):
            presum+=nums[j]
            rest=presum-k
            if rest in hmap:ans=max(ans,j-hmap[rest])
            if presum not in hmap:hmap[presum]=j
        return ans 
